subsys/shell/shell_telnet.c

```diff
@@ -196,14 +196,16 @@ static void telnet_recv(struct net_context *client,
 	}
 
 	len = net_pkt_remaining_data(pkt);
-	if (len < TELNET_MIN_MSG) {
-		LOG_DBG("Packet smaller than minimum length");
+	if (len == 0) {
```

Comment:
Will this ever happen, I mean in what case the length would be 0, did you see this in testing?

Reply:
Not really, more of a precaution from my side, if you think it's safe to remove such check I can do it.

Reply:
I checked the tcp code and the callback is not called if there is no data and connection is still up. So the length should never be 0 in this part of the code.

Reply:
Ok, I have removed the check.